server/src/main/java/org/apache/druid/guice/DruidProcessingModule.java

```diff
@@ -157,9 +157,16 @@ private void verifyDirectMemory(DruidProcessingConfig config)
       }
     }
     catch (UnsupportedOperationException e) {
+      log.debug("Checking for direct memory size is not support on this platform: %s", e);
       log.info(
-          "Could not verify that you have enough direct memory, so I hope you do! Error message was: %s",
-          e.getMessage()
+          "Unable to determine max direct memory size. If -XX:MaxDirectMemorySize is set, make sure "
```

Comment:
if above change is made then this should be something like...

"Unable to determine max direct memory size. If druid.processing.buffer.sizeBytes is explicitly set then -XX:MaxDirectMemorySize to at least 'druid.processing.buffer.sizeBytes * (druid.processing.numMergeBuffers[%,d] + druid.processing.numThreads[%,d] + 1)' or else to at least 25% of maximum heap size."